src/lib/server/mistral/mistral.ts

```diff
@@ -201,10 +201,27 @@ export class MistralAgent implements IAgent {
 
 	public async appendMessageToConversation(conversation: Conversation, prompt: AgentPrompt, streamResponse: boolean): Promise<AppendToConversationResult> {
 		if (streamResponse) {
+			// Sjekker om vi har dokumenter i vectorstore
+			let docsExists = false
+			if (conversation.vectorStoreId && typeof prompt === "string") {
+				try {
+					const docs = await getDocumentsInMistralLibrary(conversation.vectorStoreId)
+					docsExists = docs.length > 0
+				} catch (error) {
+					console.error("Feilet ved sjekk av dokumenter:", error)
+				}
+			}
+
+			// Legger kun til prefix hvis vi har dokumenter i vectorstore
+			let prefixedPrompt = prompt
+			if (docsExists) {
+				prefixedPrompt = `Always use the documents in the document library to answer the query.\n\n${prompt}`
+			}
+
```

Comment:
Enda lettere - det finnes en property i conversation som heter vectorStoreId som er satt dersom det finnes en vectorStore i samtalen. Kan bare sjekke om den er satt, så slipper man det ekstra kallet for å hente dokumenter :)
Tror vi kan sjekke litt mer på Mistral hvordan den faktisk trigger dokumentsøk også